src/pages/Chat/steps/NameAgeInput.tsx

```diff
@@ -18,86 +23,96 @@ interface Actions {
 }
 
 interface Props {
+  appear?: boolean
   onSubmit: (state?: FormValues) => void
 }
 
-const isDone = (values: FormValues) =>
-  values.firstName.length > 0 &&
-  values.lastName.length > 0 &&
-  String(values.age).length >= 2 &&
-  values.age > 0
+const handleChange = <K extends keyof Step1State>(
+  field: K,
+  chatState: ChatActions,
+) =>
+  pipe<React.ChangeEvent<HTMLInputElement>, Step1State[K], void>(
+    pathOr<string>('', ['target', 'value']),
+    (val: string | number) => chatState.setStep1Prop(field, val),
+  )
 
-export const NameAgeInput: React.SFC<Props> = ({ onSubmit }) => (
+const isDone = (
+  values: FormValues = { firstName: '', lastName: '', age: '' },
+) =>
+  pathOr(0, ['firstName', 'length'], values) > 0 &&
+  pathOr(0, ['lastName', 'length'], values) > 0 &&
+  String(pathOr(0, ['age'], values)).length >= 2 &&
+  pathOr(0, ['age'], values) > 0
+
+export const NameAgeInput: React.SFC<Props> = ({ onSubmit, appear }) => (
   <ChatContainer>
-    {({ setStep1 }) => (
+    {(chatState: ChatState & ChatActions) => (
       <Container<State, Actions>
         initialState={{ hasFocused: false }}
         actions={{ setFocused: () => (_) => ({ hasFocused: true }) }}
       >
         {(state) => (
-          <Formik<FormValues>
```

Comment:
Did `Formik` become a bad fit with the dynamic nature of this form?

Reply:
Yes :( it keeps its own state, meaning we need some kind of event emitter/subscriber system to be able to reset it a level "above" in the component tree

Reply:
Reasonable. Formik is likely better for static-type forms, like we have in the dashborad.

Reply:
yup!